### ISSUE TITLE
isTypedArray in fromPoints fails with Array of floats

### ISSUE THREAD

**Issue by vorg** [pex-gl/pex-geom#24]

**Comment by dmnsgn** [comment]
The question is broader, related to pex ecosystem: should we prohibit array of float and force typed arrays (with some utils to expand typed array easily).

**Comment by vorg** [comment]
> with some utils to expand typed array easily

There are solutions like this https://github.com/vorg/geom-builder but then you depend on using geom.count instead of geom.positions.length that might be greately padded with zeros due to last array expansion. I wonder if you could expand the underlaying ArrayBuffer but keep typed arrays tight as they are only views on underlying storage.

**Comment by vorg** [comment]
> should we prohibit array of float 

why? how do you build geometries with unknown amount of points?

**Comment by vorg** [comment]
Also there check should happen once not in a for loop

```js
 for (let i = 0; i < points.length / (isTypedArray ? 3 : 1); i++) {
    if (isTypedArray) {
      includePoint(a, points, i * 3);
    } else {
```

**Comment by vorg** [comment]
And num points should be calculated once

**Comment by dmnsgn** [comment]
The alternative check is `ArrayBuffer.isView(points)` (with the caveats it also pass new DataView test), as constructor string test is meh. We use that check in couple places (pex-renderer, pex-geom, most of your geom- packages...), should it be in pex-math utils?

Re "if check and precompute point length", benchmark says "it depends":

tl;dr, on v8:
- Comparable speeds for 8 points.
- It is slower for a Float32Array of random points below 10000 points = length 30 000 (which is often our geometry use case) and only gets faster after that amount.

<img width="782" alt="image" src="https://github.com/pex-gl/pex-geom/assets/1636460/e94f503f-fc00-4382-8cdf-9963220d5a4b">

**Comment by vorg** [comment]
Can we do tests in Safari and FF too?

**Comment by vorg** [comment]
Another argument is that ` points.length / (isTypedArray ? 3 : 1)` doesn't pass readability test.

**Comment by dmnsgn** [comment]
> Can we do tests in Safari and FF too?

Results are similar in Safari and FF (a bit less precision in browser) but 3 times slower..
<img width="2560" alt="Screenshot 2024-01-31 at 13 25 50" src="https://github.com/pex-gl/pex-geom/assets/1636460/3e8921f5-18ae-42b0-8c34-16125eab6563">
